src/components/editor/document-bar/revisions/revision-modal.tsx

```diff
@@ -0,0 +1,101 @@
+import moment from 'moment'
+import React, { useEffect, useRef, useState } from 'react'
+import { Alert, Col, ListGroup, Modal, Row, Button } from 'react-bootstrap'
+import ReactDiffViewer, { DiffMethod } from 'react-diff-viewer'
+import { Trans, useTranslation } from 'react-i18next'
+import { useParams } from 'react-router'
+import { getAllRevisions, getRevision, Revision, RevisionListEntry } from '../../../../api/revisions'
+import { UserResponse } from '../../../../api/users/types'
+import { ForkAwesomeIcon } from '../../../common/fork-awesome/fork-awesome-icon'
+import { CommonModal, CommonModalProps } from '../../../common/modals/common-modal'
+import { ShowIf } from '../../../common/show-if/show-if'
+import { RevisionButtonProps } from './revision-button'
+import './revision-modal.scss'
+import { downloadRevision, getUserDataForRevision } from './utils'
+
+export const RevisionModal: React.FC<CommonModalProps & RevisionButtonProps> = ({ show, onHide, icon, titleI18nKey, noteContent }) => {
+  const { t } = useTranslation()
+  const [revisions, setRevisions] = useState<RevisionListEntry[]>([])
+  const [selected, setSelected] = useState<number | null>(null)
+  const [selectedRevision, setSelectedRevision] = useState<Revision | null>(null)
+  const [error, setError] = useState(false)
+  const revisionAuthorListMap = useRef(new Map<number, UserResponse[]>())
+  const { id } = useParams<{ id: string }>()
+
+  useEffect(() => {
+    getAllRevisions(id).then(fetchedRevisions => {
+      fetchedRevisions.forEach(revision => {
+        const authorData = getUserDataForRevision(revision.authors)
+        revisionAuthorListMap.current.set(revision.timestamp, authorData)
+      })
+      setRevisions(fetchedRevisions)
+    }).catch(() => setError(true))
+  }, [setRevisions, setError, id])
+
+  useEffect(() => {
+    if (selected === null) {
+      return
+    }
+    getRevision(id, selected).then(fetchedRevision => {
+      setSelectedRevision(fetchedRevision)
+    }).catch(() => setError(true))
```

Comment:
should old selected revisions be cached?

Reply:
Done in 18c1cab